python-checks/src/main/java/org/sonar/python/checks/hotspots/GraphQLIntrospectionCheck.java

```diff
@@ -139,6 +149,13 @@ private static Optional<String> nameFromIdentifierOrCallExpression(Expression ex
     return Optional.ofNullable(TreeUtils.nameFromExpression(expression))
       .or(() -> TreeUtils.toOptionalInstanceOf(CallExpression.class, expression)
         .map(CallExpression::callee)
-        .map(TreeUtils::nameFromExpression));
+        .flatMap(GraphQLIntrospectionCheck::nameFromExpressionOrQualifiedExpression));
```

Comment:
I was wondering if this means that this would not work with 
```python
    gql_middlew = [
        middleware.IntrospectionMiddleware,
    ]

    GraphQLView.as_view( # OK
        'graphql',
        schema=schema,
        middleware=gql_middlew,
        backend=CustomBackend(),
        batch=True
    )
```
as the check for qualified expression is only done when encountering a CallExpression. Maybe I misunderstood this part of the code. Tell me if I am wrong.

Also this makes me think about the first implementation that could be missing the case where a Qualified expression is provided?

```python
    GraphQLView.as_view( # FN? 
        'graphql',
        schema=schema,
        middleware=(some_module.unsafemiddleware,),
        backend=CustomBackend(),
        batch=True
    )
```
Not sure about any of this but it could be nice to check. Also it is not part of this ticket and if it is too cumbersome to implement there is no need to do it.

Reply:
I will investigate this. Thanks for the tip. The first case at least should be possible without too much refactoring.

Reply:
I pushed some changes to support qualified expressions as part of the elements of the list. The changes are in this commit: cf9f1a407d2afa9522d9ebc276a1fe59be296457.